src/app/state/rewardUtils.tsx

```diff
@@ -0,0 +1,474 @@
+import { TokenInfo } from "alphadex-sdk-js";
+import { getRdtOrThrow } from "../subscriptions";
+import { StateKeyValueStoreDataRequestKeyItem } from "@radixdlt/radix-dapp-toolkit";
+
+export class ClaimComponent {
+  address: string = "";
+  dextrTokenAddress: string = "";
+  adminTokenAddress: string = "";
+  accountRewardsNftAddress: string = "";
+  orderRewardsKvsAddress: string = "";
+}
+
+export class AccountRewards {
+  accountAddress: string = "";
+  rewards: TypeRewards[] = [];
+}
+
+export class OrderRewards {
+  orderReceiptAddress: string = "";
+  orderId: number = -1;
+  orderIndex: string = "";
+  rewards: TypeRewards[] = [];
+}
+
+export class TypeRewards {
+  rewardType: string = "";
+  tokenRewards: TokenReward[] = [];
+}
+
+export class TokenReward extends TokenInfo {
+  amount: number = 0;
+}
+
+export type OrdersByTypeRewards = Map<string, Map<string, OrderTokenReward[]>>; // Map< rewardType, Map< tokenAddress , OrderTokenReward>>
+export interface OrderTokenReward {
+  orderReceiptAddress: string;
+  orderId: number;
+  orderIndex: string;
+  amount: number;
+}
+
+export function getClaimComponentFromApiData(apiData: any): ClaimComponent {
+  let claimComponent = new ClaimComponent();
+  if (apiData && apiData.items && apiData.items.length > 0) {
+    let componentApiData = apiData.items[0];
+    if (componentApiData.address && componentApiData.details?.state?.fields) {
+      claimComponent.address = componentApiData.address;
+      let stateFields = componentApiData.details.state.fields;
+      for (const fieldData of stateFields) {
+        switch (fieldData.field_name) {
+          case "dextr_token_address": {
+            claimComponent.dextrTokenAddress = fieldData.value;
+            break;
+          }
+          case "admin_token_address": {
+            claimComponent.adminTokenAddress = fieldData.value;
+            break;
+          }
+          case "account_rewards_nft_manager": {
+            claimComponent.accountRewardsNftAddress = fieldData.value;
+            break;
+          }
+          case "order_rewards": {
+            claimComponent.orderRewardsKvsAddress = fieldData.value;
+            break;
+          }
+        }
+      }
+    }
+  }
+  return claimComponent;
+}
+
+export function getTypeRewards(
```

Comment:
This function has a lot of duplicate code. To make it more maintainable and readable, I suggest to do this instead:

```ts
const typeRewards: TypeRewards[] = [
  ...accountsRewards.flatMap((accountReward) => accountReward.rewards),
  ...ordersRewards.flatMap((orderRewards) => orderRewards.rewards),
];
typeRewards.forEach((typeReward) => { /* duplicate logic */ }
```

The full function would then look like this (Disclaimer: I did simplify this inside my codeeditor without any testing. Please test before using):
```ts
export function getTypeRewards(
  accountsRewards: AccountRewards[],
  ordersRewards: OrderRewards[],
  tokensList: TokenInfo[]
): TypeRewards[] {
  let tokensMap: Map<string, TokenInfo> = new Map();
  tokensList.forEach((tokenInfo) =>
    tokensMap.set(tokenInfo.address, tokenInfo)
  );
  let typeRewardsMap: Map<string, Map<string, TokenReward>> = new Map();
  const typeRewards: TypeRewards[] = [
    ...accountsRewards.flatMap((accountReward) => accountReward.rewards),
    ...ordersRewards.flatMap((orderRewards) => orderRewards.rewards),
  ];
  typeRewards.forEach((typeReward) => {
    let existingTypeRewardTokens = typeRewardsMap.get(typeReward.rewardType);
    if (!existingTypeRewardTokens) {
      existingTypeRewardTokens = new Map<string, TokenReward>();
    } else {
      existingTypeRewardTokens = new Map(existingTypeRewardTokens);
    }
    typeReward.tokenRewards.forEach((tokenReward) => {
      let existingTokenReward = existingTypeRewardTokens!.get(
        tokenReward.address
      );
      if (!existingTokenReward) {
        let tokenInfo = tokensMap.get(tokenReward.address);
        if (!tokenInfo) {
          tokenInfo = new TokenInfo(
            tokenReward.address,
            "?",
            "Unknown Token",
            "/unknown-token-icon.svg"
          );
        }
        existingTokenReward = { ...tokenInfo, amount: tokenReward.amount };
      } else {
        existingTokenReward = { ...existingTokenReward };
        existingTokenReward.amount += tokenReward.amount;
      }
      existingTypeRewardTokens!.set(
        existingTokenReward.address,
        existingTokenReward
      );
    });
    typeRewardsMap.set(typeReward.rewardType, existingTypeRewardTokens);
  });
  let result: TypeRewards[] = [];
  typeRewardsMap.forEach((tokenRewardsMap, rewardType) => {
    result.push({
      rewardType,
      tokenRewards: Array.from(tokenRewardsMap.values()),
    });
  });
  return result;
}
```

Reply:
Feel free to ignore.

Reply:
Would rather do this in a separate issue when we have more time. We know the existing one works so lets go with that for now.